python/pyspark/errors/exceptions/captured.py

```diff
@@ -65,8 +65,15 @@ def __str__(self) -> str:
         assert SparkContext._jvm is not None
 
         jvm = SparkContext._jvm
-        sql_conf = jvm.org.apache.spark.sql.internal.SQLConf.get()
-        debug_enabled = sql_conf.pysparkJVMStacktraceEnabled()
+
+        # SPARK-42752: default to True to see issues with initialization
+        debug_enabled = True
+        try:
+            sql_conf = jvm.org.apache.spark.sql.internal.SQLConf.get()
+            debug_enabled = sql_conf.pysparkJVMStacktraceEnabled()
+        except BaseException:
```

Comment:
Maybe can we capture the Exception more specifically for `pyspark.sql.utils.IllegalArgumentException` instead of capturing `BaseException` ??

Reply:
I advocate for keeping the likelihood of an unhelpful unprintable exception during initialization to the minimum. I would not want to revisit the issue for other runtime exceptions.

Reply:
Sounds good. Thanks for working on this!